.travis/global-namespace-check.sh

```diff
@@ -1,11 +1,11 @@
 #!/bin/bash
 
-if [[ $(nm -g liboqs.a | grep ' T ' | grep -E -v -i ' T [_]?[OQS|ntru|picnic|Keccak]') ]];
+if [[ $(nm -g liboqs.a | grep ' T ' | grep -E -v -i ' T [_]?(OQS|ntru|picnic|Keccak|.*SIKEp503|.*SIDHp503|.*SIKEp751|.*SIDHp751|.*shake128|.*shake256|rand_bytes|print_hex|cpu_supports|uint64_from_char_array|uint64_to_char_array|_ntt_double|_rec|_aligned_alloc|_aligned_free)') ]];
 then
 	tput setaf 1;
 	echo "Code contains the following non-namespaced global symbols; see https://github.com/open-quantum-safe/liboqs/wiki/Coding-conventions for function naming conventions.";
 	tput sgr 0
-	nm -g liboqs.a | grep ' T ' | grep -E -v -i ' T [_]?[OQS|ntru|picnic|Keccak]'
+  nm -g liboqs.a | grep ' T ' | grep -E -v -i ' T [_]?(OQS|ntru|picnic|Keccak|.*SIKEp503|.*SIDHp503|.*SIKEp751|.*SIDHp751|.*shake128|.*shake256|rand_bytes|print_hex|cpu_supports|uint64_from_char_array|uint64_to_char_array|_ntt_double|_rec)'
```

Comment:
I'd be worried about a few of these being popular names, including print_hex, _ntt_double, and _rec.  What's the motivation for including these?

Also, is there a reason that line 8's regex is slightly different than line 3's?  Perhaps we should make the regex a variable and then use that variable on both lines.

Reply:
I refactored the regex into the shell variable REGEX. Not sure what was the initial motivation of disabling `print_hex`, `_ntt_double`, and `_rec`, so I kept them. @smashra any idea?